SQLQuery1.sql

```diff
@@ -0,0 +1,212 @@
+
+alter table users
+alter column user_name varchar(16) not null
+
+alter table users
+alter column pass_word varchar(10) not null
+
+alter table project
+add foreign key (category_id) references categories(id);
+
+alter table project
+add foreign key (company_id) references companies(id);
+
+alter table project
+alter column project_spend int
+
+alter table project
+alter column project_variance int
+
+alter table project
+alter column revenue_recognised int
+
+alter table users
+alter column birthday date
+
+alter table users
+alter column image_url varchar
+
+
+-- CHèn dữ liệu vào bảng
+
+insert into users(
+user_name,
+email,
+pass_word,
+birthday,
+is_active,
+created_at,
+updated_at
+)
+values (
+'Dao van B',
+'yyyyyyyyyyy@gmail.com',
+'12345678',
+'2002/12/23',
+1,
+DEFAULT,
+'2021/12/23'
+)
+
+Insert into companies(
+company_name,
+company_code,
+created_at,
+updated_at
+)
+values (
+'monstar-lab',
+'1234',
+default,
+'2011-11-01'
+)
+
+insert into Project (
+project_name,
+category_id,
+project_spend,
+project_variance,
+revenue_recognised,
+created_at,
+updated_at,
+company_id
+)
+values (
+'project_4',
+2,
+132,
+123,
+23,
+default,
+'2021/05/30',
+2
+)
+
+insert into Project_user (
+project_id,
+user_id,
+created_at,
+updated_at
+)
+values (
+10,
+3,
+default,
+'2020/12/05'
+)
+
+--Viết lệnh sql để có thể lấy thông tin những bản ghi của projects 
+--và số lượng user của mỗi projects đó (count user)
+select project.id,project_name,project_spend,Project.project_variance,Project.category_id, count(users.id) as "count_user" from Project
+left join project_user on Project.id= Project_user.project_id
+left join users on Project_user.user_id=USERs.id
+group by project.id,project_name,project_spend,Project.project_variance,Project.category_id
+
+
+--viết lệnh sql để lấy ra danh sách các project của company có company_name = “monstar-lab” 
+
+select * from Project
+inner join companies
+on Project.company_id=companies.id
+where company_name = 'monstar-lab'
+
+--viết lệnh sql lấy ra danh sách các công ty có project có project_spend > 100
+
+select companies.company_name from companies
+inner join Project
+on Project.company_id=companies.id
+where project.project_spend>100
+
+--viết lệnh sql để lấy ra thông tin của user tham gia vào projects
+select distinct USERs.id,user_name,email,pass_word,birthday,image_url,USERs.created_at,USERs.updated_at from users 
+inner join Project_user
+on users.id=Project_user.user_id
+inner join Project
+on Project_user.project_id=Project.id
+
+--lấy ra danh sách project mà có số lượng user tham gia > 10 , sắp xếp số lượng user tham gia tăng dần
+
+select project_name,
+count(users.id) as "count_user" 
+from Project
+left join project_user on Project.id= Project_user.project_id
+left join users on Project_user.user_id=USERs.id
+group by project_name
+having count(users.id)>10
+order by count(users.id) asc
+
+
+--Xoá project mà chưa có user nào tham gia
+use bang_test
+delete from project 
+where project_name = (
```

Comment:
có thể viết ngắn gọn dùng WHERE NOT EXISTS(SELECT project_id FROM project_users WHERE projects.id = project_users.project_id)